querysync/java/com/google/idea/blaze/qsync/project/BuildGraphData.java

```diff
@@ -546,16 +559,22 @@ public ImmutableSet<QuerySyncLanguage> getTargetLanguages(ImmutableSet<Label> ta
    *     {@link #getDependencyTrackingBehaviors(Label)} of the targets given.
    */
   public Optional<RequestedTargets> computeRequestedTargets(ImmutableSet<Label> projectTargets) {
-    ImmutableSet<Label> externalDeps =
-        projectTargets.stream()
-            .filter(
-                t ->
-                    getDependencyTrackingBehaviors(t).stream()
-                        .anyMatch(b -> b.shouldIncludeExternalDependencies))
-            .map(this::getTransitiveExternalDependencies)
-            .flatMap(Set::stream)
-            .collect(toImmutableSet());
-
-    return Optional.of(new RequestedTargets(projectTargets, externalDeps));
+    final var externalDeps = new LinkedHashSet<Label>();
+    final var seen = new HashSet<>(projectTargets);
+    final var queue = new ArrayDeque<Label>(projectTargets);
+    while (!queue.isEmpty()) {
```

Comment:
Thumbs up for removing the unnecessary stream. Let's not pretend that Java is a FP language.